classes/local/api.php

```diff
@@ -185,7 +185,9 @@ private function add_postname($storedfile, $key) {
             throw new \moodle_exception('needphp55orhigher', 'tool_opencast');
         }
 
-        $filename = $storedfile->get_source();
+        $sourse = @unserialize($storedfile->get_source());
+        $filename = $sourse ? $sourse : $storedfile->get_source();
+        is_string($filename) ? $filename = $filename : $filename = $filename->source ;
```

Comment:
This last line is very hard to read and, actually, I doubt that it does the right thing. The `?:`-operator is usually only used to produce an expression, but here, the expression is not used for anything. Also, the part between `?` and `:` here is essentially a no-op, i.e., nothing is changed. I'd suggest rewriting this line using an `if`-statement, focusing on the relevant branch.